basicdataset/src/test/java/ai/djl/basicdataset/PennTreebankTextTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.basicdataset;
+
+import ai.djl.basicdataset.nlp.PennTreebankText;
+import ai.djl.basicdataset.utils.TextData.Configuration;
+import ai.djl.ndarray.NDManager;
+import ai.djl.repository.Repository;
+import ai.djl.training.dataset.Dataset;
+import ai.djl.training.dataset.Record;
+import ai.djl.translate.TranslateException;
+import java.io.IOException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class PennTreebankTextTest {
+
+    private static final int EMBEDDING_SIZE = 15;
+
+    @Test
+    public void testPennTreebankTextTrainLocal() throws IOException, TranslateException {
```

Comment:
I notice you have three tests which are exactly the same besides the usage. Rather than copying and pasting, it is better to loop through the different usages. The technically correct way is to use TestNg parameterized tests, but it is probably easier to just have a `for (Usage usage : new Usage[]{Usage.TRAIN, Usage.VALIDATION, Usage.TEST})` loop.